### ISSUE TITLE
Route calculation using relations containing only stops

### ISSUE THREAD

**Issue by contrapunctus-1** [osmandapp/Osmand#8632]
OsmAnd v3.5.5

When routing using relations containing only platforms (which is a faster and easier way to create PT routes - it is being formalized into a [new schema](https://wiki.openstreetmap.org/wiki/Proposed_features/Simpler_public_transport_route_relations)), the route between stops is shown as a direct line from one stop to another, instead of following the road. (I'm not sure if the distance and ETA is also calculated based on that.)

Please consider drawing the route between stops by following the road.

**Comment by vshcherb** [comment]
I saw PTv3 and I think it is probably too much hassle for display programs, I hope we will stay with v2 and instead of v3 we will jump to something automatic transformations from GTFS (which will be published regularly in Europe) to OSM, obviously with automatic conversion it there will be ways stored.

**Comment by aceman444** [comment]
Really, having any displaying app calculate the route automatically seems too error prone.

**Comment by sonora** [comment]
Yes, where you use PT for transport, what counts is where you can get on and off, so the platforms connected by straight lines are sufficient. But where you use it for e.g. sightseeing, i.e. the when the exact route is significant, there is no reliable way to derive it from just the stops.

**Comment by scaidermern** [comment]
> Yes, where you use PT for transport, what counts is where you can get on and off, so the platforms connected by straight lines are sufficient.

I fully agree. Usually I want to get from A to B. Neither does it matter nor do I care which route the bus/tram/whatever will take between stops A and B (except for the resulting duration, of course).

**Comment by sonora** [comment]
Just happen to come across this once more. I believe we should close this issue based on technical impossibility?

I do not think we should provide a route line between stops based on guesses how thing might be? As is, at least the users have a visual indication that there is no data, and they could take their own guess?

**Comment by aceman444** [comment]
@sonora actually the report is true, for routing purposes the bus platforms in the relation are enough as you can only enter/exit the bus on those platforms. The route inbetween is not needed, only for some nice visual effect and also for previewing bus routes (when you enable routes in OsmAnd) without making OsmAnd calculate a route.
So maybe sometimes the author does not know the exact route and only put platforms (stops) into the relation for now. Having OsmAnd support this would be an additional feature, even when such a relation isn't valid per PTv2 schema right now (but it has enough data for OsmAnd). But OSM contains a ton of non-conformant data.
